app/gui2/src/components/GraphEditor/GraphNodes.vue

```diff
@@ -32,6 +37,13 @@ function moveNode(movedId: ExprId, delta: Vec2) {
 function hoverNode(id: ExprId | undefined) {
   if (selection != null) selection.hoveredNode = id
 }
+
+const uploadingFiles = computed<[FileName, File][]>(() => {
+  const currentStackItem = projectStore.executionContext.getStackTop()
+  return [...projectStore.awareness.allUploads()].filter(
+    ([_name, file]) => file.stackItem == currentStackItem,
```

Comment:
This comparison is incorrect. It checks for reference equality, which will most of the time be false. When you pop and push to a stack, the current stack item might be identical, but will be a separate object instance. You actually need to perform a deep comparison by implementing your own `stackItemEquals` function.

Reply:
Fixed